tools/jdbc/src/test/java/org/duckdb/test/TestDuckDBJDBC.java

```diff
@@ -3211,13 +3219,36 @@ public static void test_union() throws Exception {
 
 	public static void test_list() throws Exception {
 		try (Connection connection = DriverManager.getConnection("jdbc:duckdb:");
-			 PreparedStatement statement = connection.prepareStatement("select [1]")) {
-			ResultSet rs = statement.executeQuery();
-			assertTrue(rs.next());
-			assertEquals(rs.getObject(1), "[1]");
+			 Statement statement = connection.createStatement()) {
+			try (ResultSet rs = statement.executeQuery("select [1]")) {
+				assertTrue(rs.next());
+				assertEquals(arrayToList(rs.getArray(1)), singletonList(1));
+			}
+			try (ResultSet rs = statement.executeQuery("select unnest([[1], [42, 69]])")) {
+				assertTrue(rs.next());
+				assertEquals(arrayToList(rs.getArray(1)), singletonList(1));
+				assertTrue(rs.next());
+				assertEquals(arrayToList(rs.getArray(1)), Arrays.asList(42, 69));
+			}
+			try (ResultSet rs = statement.executeQuery("select unnest([[[42], [69]]])")) {
```

Comment:
I would like to see some more tests with nested lists, lists of strings, empty lists, etc. Perhaps we could add the test with `test_all_types` while excluding the columns/types not yet supported? That should cover a lot of the potential edge cases here.

Reply:
Okay, added the test_all_types() test - not sure if this one is still worth keeping?